arangod/VocBase/Methods/Indexes.cpp

```diff
@@ -712,105 +750,153 @@ futures::Future<arangodb::Result> Indexes::drop(LogicalCollection& collection,
-                                    trxOpts);
+  return std::make_unique<SingleCollectionTransaction>(
+      transaction::V8Context::createWhenRequired(collection.vocbase(), origin,
+                                                 false),
+      collection, AccessMode::Type::EXCLUSIVE, trxOpts);
 #else
-    SingleCollectionTransaction trx(
-        transaction::StandaloneContext::create(collection.vocbase(), origin),
-        collection, AccessMode::Type::EXCLUSIVE, trxOpts);
+  return std::make_unique<SingleCollectionTransaction>(
+      transaction::StandaloneContext::create(collection.vocbase(), origin),
+      collection, AccessMode::Type::EXCLUSIVE, trxOpts);
 #endif
-    Result res = co_await trx.beginAsync();
+}
 
-    if (!res.ok()) {
-      events::DropIndex(collection.vocbase().name(), collection.name(), "",
-                        res.errorNumber());
-      co_return res;
-    }
+template<typename IndexSpec>
+requires std::is_same_v<IndexSpec, IndexId> or
+    std::is_same_v<IndexSpec, velocypack::Slice>
+        futures::Future<arangodb::Result> Indexes::dropDBServer(
+            LogicalCollection& col, IndexSpec indexSpec) {
+  TRI_ASSERT(!ServerState::instance()->isCoordinator());
+  READ_LOCKER(readLocker, col.vocbase()._inventoryLock);
 
-    LogicalCollection* col = trx.documentCollection();
-    res = co_await getHandle(trx.resolver(), &trx);
-    if (!res.ok()) {
-      co_return res;
-    }
+  auto trx = createTrxForDrop(col);
 
-    std::shared_ptr<Index> idx = collection.lookupIndex(iid);
-    if (!idx || idx->id().empty() || idx->id().isPrimary()) {
-      events::DropIndex(collection.vocbase().name(), collection.name(),
-                        std::to_string(iid.id()),
-                        TRI_ERROR_ARANGO_INDEX_NOT_FOUND);
-      co_return {TRI_ERROR_ARANGO_INDEX_NOT_FOUND};
-    }
-    if (!idx->canBeDropped()) {
-      events::DropIndex(collection.vocbase().name(), collection.name(),
-                        std::to_string(iid.id()), TRI_ERROR_FORBIDDEN);
-      co_return {TRI_ERROR_FORBIDDEN};
-    }
+  auto beginRes = co_await trx->beginAsync();
+  auto& collection = *trx->documentCollection();
 
-    res = col->dropIndex(idx->id());
+  if (!beginRes.ok()) {
+    events::DropIndex(collection.vocbase().name(), collection.name(), "",
+                      beginRes.errorNumber());
+    co_return beginRes;
+  }
+
+  auto indexIdRes = co_await std::invoke(
+      overload{
+          [](IndexId indexId) -> futures::Future<ResultT<IndexId>> {
+            co_return indexId;
+          },
+          [&](velocypack::Slice indexArg) -> futures::Future<ResultT<IndexId>> {
+            auto res = co_await getHandle(collection, indexArg, trx->resolver(),
+                                          trx.get());
+            if (!res.ok()) {
+              co_return std::move(res).result();
+            }
+            auto&& [iid, name] = res.get();
+            co_return iid;
```

Comment:
In this case, it's not the same: The transaction is passed to getHandle. In the coordinator case, this will eventually open its own (read) transaction somewhere down the line.
I'm not sure about the impact, I just assumed it was this way for a reason and didn't want to change it.

Reply:
Good point.